test/integration/apiserver/field_validation_test.go

```diff
@@ -0,0 +1,2703 @@
+
+	_, client, closeFn := setup(t)
+	defer closeFn()
+
+	bodyJSON := `
+	{
+		"apiVersion": "apps/v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "%s",
+			"labels": {"app": "nginx"}
+		},
+		"spec": {
+			"unknown1": "val1",
+			"unknownDupe": "valDupe",
+			"unknownDupe": "valDupe2",
+			"paused": true,
+			"paused": false,
+			"selector": {
+				"matchLabels": {
+					"app": "nginx"
+				}
+			},
+			"template": {
+				"metadata": {
+					"labels": {
+						"app": "nginx"
+					}
+				},
+				"spec": {
+					"containers": [{
+						"name":  "nginx",
+						"image": "nginx:latest",
+						"unknownNested": "val1",
+						"imagePullPolicy": "Always",
+						"imagePullPolicy": "Never"
+					}]
+				}
+			}
+		}
+	}
+		`
+
+	bodyYAML :=
+		`apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: %s
+  labels:
+    app: nginx
+spec:
+  unknown1: val1
+  unknownDupe: valDupe
+  unknownDupe: valDupe2
+  paused: true
+  paused: false
+  selector:
+    matchLabels:
+      app: nginx
+  template:
+    metadata:
+      labels:
+        app: nginx
+    spec:
+      containers:
+      - name: nginx
+        image: nginx:latest
+        unknownNested: val1
+        imagePullPolicy: Always
+        imagePullPolicy: Never`
+	var testcases = []struct {
+		name                   string
+		bodyBase               string
+		opts                   metav1.CreateOptions
+		contentType            string
+		strictDecodingErrors   []string
+		strictDecodingWarnings []string
+	}{
+		{
+			name: "post-strict-validation",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Strict",
+			},
+			bodyBase: bodyJSON,
+			strictDecodingErrors: []string{
+				`unknown field "unknown1"`,
+				`unknown field "unknownDupe"`,
+				`duplicate field "paused"`,
+				// note: fields that are both unknown
+				// and duplicated will only be detected
+				// as unknown for typed resources.
+				`unknown field "unknownNested"`,
+				`duplicate field "imagePullPolicy"`,
+			},
+		},
+		{
+			name: "post-warn-validation",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Warn",
+			},
+			bodyBase: bodyJSON,
+			strictDecodingWarnings: []string{
+				`unknown field "unknown1"`,
+				`unknown field "unknownDupe"`,
+				`duplicate field "paused"`,
+				// note: fields that are both unknown
+				// and duplicated will only be detected
+				// as unknown for typed resources.
+				`unknown field "unknownNested"`,
+				`duplicate field "imagePullPolicy"`,
+			},
+		},
+		{
+			name: "post-ignore-validation",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Ignore",
+			},
+			bodyBase: bodyJSON,
+		},
+		{
+			name:     "post-default-ignore-validation",
+			bodyBase: bodyJSON,
+		},
+		{
+			name: "post-strict-validation-yaml",
+			opts: metav1.CreateOptions{
+				FieldValidation: "Strict",
+			},
+			bodyBase:    bodyYAML,
+			contentType: "application/yaml",
+			strictDecodingErrors: []string{
+				`line 10: key "unknownDupe" already set in map`,
```

Comment:
See my [comment](https://github.com/kubernetes/kubernetes/pull/105916#discussion_r743044300) for why these yaml strict errors look different.